signac/core/collection_mongodb.py

```diff
@@ -44,10 +35,17 @@ def _sync(self):
         """Write the data from Mongo-database."""
         data = self.to_base()
         data_to_insert = {self._key: self._name, 'data': data}
-        self._collection.replace_one({self._key: self._name}, data_to_insert, True)
+        try:
+            self._collection.replace_one({self._key: self._name}, data_to_insert, True)
+        except self._errors.InvalidDocument as err:
+            raise TypeError(str(err))
 
+    def __deepcopy__(self, memo):
+        return type(self)(collection=self._collection, name=self._name, data=self.to_base(),
```

Comment:
For a proper deepcopy, you have to deepcopy `self._collection` and `self._name` as well.

Reply:
I tried to do that but `deepcopy(self._collection)` raise `TypeError`.

Reply:
See comment above.